tests/transactional/host_config.pm

```diff
@@ -1,24 +1,37 @@
 # SUSE's openQA tests
 #
-# Copyright 2020-2023 SUSE LLC
+# Copyright 2020-2024 SUSE LLC
 # SPDX-License-Identifier: FSFAP
 #
 # Package: transactional-update
-# Summary: Host configuration operations (e.g. disable grub timeout,
-#              kernel params, etc)
+# Summary: Host configuration operations
+# * Configure virtio-terminal
+# * Configure bootloader (disable timeout, extra kernel parameters)
+# * Install SUSE CA certificates, if applicable
 #
 # Maintainer: QE-C team <qa-c@suse.de>
 
 use Mojo::Base qw(consoletest);
 use testapi;
 use transactional qw(process_reboot);
 use bootloader_setup qw(change_grub_config);
-use utils qw(ensure_ca_certificates_suse_installed zypper_call);
+use utils qw(ensure_ca_certificates_suse_installed zypper_call ensure_serialdev_permissions);
 use version_utils qw(is_bootloader_grub2 is_bootloader_sdboot);
+use serial_terminal qw(prepare_serial_console);
 
 sub run {
     select_console 'root-console';
 
+    # Ensure testuser (UID 1000) is present for the user-virtio-terminal
+    # this needs to happen before ensure_serialdev_permissions
+    if (script_run("getent passwd 1000") != 0) {
+        assert_script_run("useradd -u 1000 -m $testapi::username");
+        assert_script_run("echo '$testapi::username:$testapi::password' | chpasswd");
+    }
```

Comment:
I hoped that this would be in `prepare_serial_console()`. I suppose you have a reason not to be there. Is it breaking other installations?

Reply:
Honestly - Because user 1000 is typically created everywhere during in the installation. It is the exception for MicroOS/SLEM that this user doesn't exist.

So I think as the exception for the rule it can also just reside in the `host_config` step of that particular system.

Reply:
We can move it to `prepare_serial_console`, but everywhere I saw that `ensure_serialdev_permissions` is run before that step, and `ensure_serialdev_permissions` requires UID 1000 already.

So we would also need to switch the order of those two subroutines everywhere.

Reply:
@pevik I moved it now to the library, please check if this is what you intended.

If this is ok with you, I will need to add more VRs for other flavors as well.

Reply:
Thank you. Is there a reason not to add `ensure_testuser_present()` into beginning of `prepare_serial_console()`? Because even this is more readable when put into it's own function, it's: for virtio console you need `prepare_serial_console()` and sometimes also `ensure_testuser_present()`. I hoped, that everything needed would be called from `prepare_serial_console()`, thus new products in the future add just this function.

Please make a backup of the current code and if adding  `ensure_testuser_present` into beginning of `prepare_serial_console` breaks anything, I'll give up and let the current version go. I can even help you with the testing other products (Tumbleweed, SLES).

Reply:
I put `ensure_testuser_present` in `prepare_serial_console` after all the skip steps. Adding a new user if not present should only happen if we really need it, and if we're skipping the preparation step then this can also be skipped.

We can move it up if you like, that was just me being conservative with additional code/modification that is being used in practically almost all test runs we have on OSD and O3.